### ISSUE TITLE
Question: Computing what percentage one duration is of another

### ISSUE THREAD

**Issue by ajxs** [moment/moment#5042]
My specific use case requires using Moment to find what percentage one duration is of another.
e.g. 3 months being roughly 25% of one year.
Given 2 periods represented by 2 start and 2 end dates converted to durations, what is the best way to compute the percentage one is of the other?
This is what I'm doing currently:

```
const firstPeriodLength = moment.duration(firstEndDate.diff(firstStartDate));
const secondPeriodLength = moment.duration(secondEndDate.diff(secondStartDate));

// BigNumber.js is used for better accuracy.
const factor = new BigNumber(firstPeriodLength.as("milliseconds"))
	.dividedBy(secondPeriodLength.as("milliseconds"));
```

Excuse the use of Typescript here.
This seems to yield a usable result, but I'm concerned with the overall accuracy of such a method.
Is there a better way to do this? This seems pretty good, but I need it to be as accurate as I can get it.
If anyone could advise on the best way to achieve this, I'd be very grateful!
Thank you so much.

**Comment by kylekatarnls** [comment]
Hello,

What about:
```js
const firstStartDate = moment('2019-03-17');
const secondStartDate = moment('2019-03-28');
const now = moment();
console.log((now - firstStartDate ) / (secondStartDate  - firstStartDate)); // 0.15
```
This magic is available thanks to `valueOf()`.

PS: `const` is ES6, not TypeScript-specific.

**Comment by ajxs** [comment]
Thanks for your response. You're correct, I had just copied and adapted that example straight from my Typescript source without checking whether it had type annotations or other Typescript specifics.
I'm trying to track down the source of what I think are some truncation errors in some very precise calculations, part of which requires working this out.
I tested this using `moment(...).unix()` to get the raw epoch values for the start and end values and using those to get the lengths directly, and got the same value. So I'm starting to think this is fairly accurate.
